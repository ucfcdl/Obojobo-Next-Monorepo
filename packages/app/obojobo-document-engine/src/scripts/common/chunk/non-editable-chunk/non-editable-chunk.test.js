import React from 'react'
import renderer from 'react-test-renderer'

import NonEditableChunk from './index'

describe('NonEditableChunk', () => {
	test('renders NonEditableChunk component with className, indent, and children', () => {
		const componentContents = (
			<NonEditableChunk className="testClass" indent={2}>
				Stuff here
			</NonEditableChunk>
		)
		const component = renderer.create(componentContents)
		const tree = component.toJSON()

		expect(tree).toMatchSnapshot()
	})

	test('renders NonEditableChunk component with hangingIndent', () => {
		const componentContents = (
			<NonEditableChunk className="testClass" hangingIndent={true}>
				Stuff here
			</NonEditableChunk>
		)
		const component = renderer.create(componentContents)
		const tree = component.toJSON()

		expect(tree).toMatchSnapshot()
	})

	test('renders NonEditableChunk component without className, indent, and children', () => {
		const componentContents = <NonEditableChunk />
		const component = renderer.create(componentContents)
		const tree = component.toJSON()

		expect(tree).toMatchSnapshot()
	})
})
