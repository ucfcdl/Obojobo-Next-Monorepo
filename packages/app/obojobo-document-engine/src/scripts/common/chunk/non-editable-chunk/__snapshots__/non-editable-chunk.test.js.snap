// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NonEditableChunk renders NonEditableChunk component with className, indent, and children 1`] = `
<div
  className="non-editable-chunk testClass"
  contentEditable={false}
  data-indent={2}
>
  Stuff here
</div>
`;

exports[`NonEditableChunk renders NonEditableChunk component with hangingIndent 1`] = `
<div
  className="non-editable-chunk testClass"
  contentEditable={false}
  data-hanging-indent={true}
  data-indent={0}
>
  Stuff here
</div>
`;

exports[`NonEditableChunk renders NonEditableChunk component without className, indent, and children 1`] = `
<div
  className="non-editable-chunk"
  contentEditable={false}
  data-indent={0}
/>
`;
